tests/testsuite/update.rs

```diff
@@ -2096,3 +2095,466 @@ fn update_breaking_specific_packages() {
+        .with_stderr("")
+        .run();
+
+    // Spec version not matching our current dependencies
+    p.cargo("update -Zunstable-options --breaking incompatible@2.0.0")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_stderr("")
+        .run();
+
+    // Spec source not matching our current dependencies
+    p.cargo("update -Zunstable-options --breaking https://alternative.com#incompatible@1.0.0")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_stderr("")
+        .run();
+
+    // Accepted spec
+    p.cargo("update -Zunstable-options --breaking incompatible@1.0.0")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_stderr("")
+        .run();
+
+    // Accepted spec, full format
+    Package::new("incompatible", "3.0.0").publish();
+    p.cargo("update -Zunstable-options --breaking https://github.com/rust-lang/crates.io-index#incompatible@2.0.0")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_stderr("")
+        .run();
+
+    // Spec matches a dependency that will not be upgraded
+    p.cargo("update -Zunstable-options --breaking compatible@1.0.0")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_stderr("")
+        .run();
+
+    // Non-existing versions
+    p.cargo("update -Zunstable-options --breaking incompatible@9.0.0")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_stderr("")
+        .run();
+
+    p.cargo("update -Zunstable-options --breaking compatible@9.0.0")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        .with_stderr("")
+        .run();
+}
+
+#[cargo_test]
+fn update_breaking_spec_version_transitive() {
+    Package::new("dep", "1.0.0").publish();
+    Package::new("dep", "1.1.0").publish();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name  =  "foo"
+                version  =  "0.0.1"
+                edition  =  "2015"
+                authors  =  []
+
+                [dependencies]
+                dep  =  "1.0"
+                bar = { path = "bar", version = "0.0.1" }
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .file(
+            "bar/Cargo.toml",
+            r#"
+                [package]
+                name  =  "bar"
+                version  =  "0.0.1"
+                edition  =  "2015"
+                authors  =  []
+
+                [dependencies]
+                dep  =  "1.1"
+            "#,
+        )
+        .file("bar/src/lib.rs", "")
+        .build();
+
+    p.cargo("generate-lockfile").run();
+
+    Package::new("dep", "1.1.1").publish();
+    Package::new("dep", "2.0.0").publish();
+
+    // Will upgrade the direct dependency
+    p.cargo("update -Zunstable-options --breaking dep@1.0")
+        .masquerade_as_nightly_cargo(&["update-breaking"])
+        // FIXME: Should upgrade a dependency here.
```

Comment:
Let's have a single place to track remaining work, your task list, rather than having multiple backlogs, with one being harder to track (the source) than the other (issues)

Reply:
These FIXMEs were just intermediate ones, and are gone in the final PR.

But task list added here: https://github.com/rust-lang/cargo/issues/12425#issuecomment-2186198258